pkg/kubelet/pod/pod_manager.go

```diff
@@ -330,7 +330,10 @@ func (pm *basicManager) IsMirrorPodOf(mirrorPod, pod *v1.Pod) bool {
 	if !ok {
 		return false
 	}
-	return hash == getPodHash(pod)
+	if hash == getPodHash(pod) {
+		return true
+	}
+	return hash == getPodLegacyHash(pod)
```

Comment:
Please add a test case for this.

Reply:
will address your comments after @yujuhong comment is addressed :)